app/Http/Middleware/HeadersMiddleware.php

```diff
@@ -10,19 +10,17 @@
 class HeadersMiddleware
 {
     public const X_TRACEID = 'x-traceid';
-    private TraceIdGenerator $traceIdGenerator;
 
-    public function __construct(TraceIdGenerator $traceIdGenerator)
+    public function __construct()
     {
-        $this->traceIdGenerator = $traceIdGenerator;
     }
 
     public function handle(Request $request, Closure $next): Response
     {
         $traceId = $request->headers->get(self::X_TRACEID);
 
         if (is_null($traceId)) {
-            $traceId = $this->traceIdGenerator->generate();
+            $traceId = TraceIdGenerator::generate();
```

Comment:
Not sure if making this method static is the best way to go. We start incentivating using this method anywhere with an implicit dependency, this may lead to bad designs.